WapTenantPublicAPI.psm1

```diff
@@ -549,6 +549,192 @@ function GetWAPSubscriptionQuota {
+        PS C:\>Get-WAPSubscriptionQuota -Subscription $Sub
+    #>
+[CmdletBinding()]
+Param(
+    [Parameter(
+        Mandatory=$False,
+        ValueFromPipeline=$True,
+        ValueFromPipelineByPropertyName=$True,
+        Position=0
+    )]
+    [System.Management.Automation.PSTypeName('WAP.Subscription')] 
+    $Subscription = (Get-WAPSubscription)
+)
+process{
+    ForEach ($Sub in $Subscription){
+        Select-WAPSubscription -Subscription $Sub
+        $VMs = Get-WAPVM 
+        $Cores = $Vms.CpuCount | Measure-Object -Sum 
+        $Memory = $Vms.Memory | Measure-Object -Sum 
+
+        $Services = $Sub.Services | Select BaseQuotaSettings
+
+        ForEach($Service in $Services){
+            $BaseQuotaSettings = $Service.BaseQuotaSettings
+            ForEach($Setting in $BaseQuotaSettings){ 
+                if($Setting.Key -eq "Clouds"){
+                    [xml]$Data = $Setting.Value 
+                    $Output = $Data.Clouds.Cloud.Quota
+                    $Object = New-Object -Type PsObject -Property @{
+                    SubscriptionName=$Sub.SubscriptionName;
+                    CurrentVms=$Cores.Count;
+                    CurrentCores=$Cores.Sum;
+                    CurrentMemory=$Memory.Sum;
+                    QuotaRoleVMCount=$Output.RoleVMCount;
+                    QuotaMemberVMCount=$Output.MemberVMCount;
+                    QuotaRoleCPUCount=$Output.RoleCPUCount;
+                    QuotaMemberCPUCount=$Output.MemberCPUCount;
+                    QuotaMemberMemory=$Output.MemberMemoryMB;
+                    QuotaRoleMemory=$Output.RoleMemoryMB;
+                    QuotaRoleStorageGB=$Output.RoleStorageGB;
+                    QuotaMemberStorageGB=$Output.MemberStorageGB
+                    }
+                    $Object.PSObject.TypeNames.Insert(0,'WAP.QUOTA')
+                    Write-Output $Object
+                }
+            }
+        }
+    } 
+}
+}
+
+function Get-WAPVMRoleQuotaRequest{
+    <#
+    .SYNOPSIS
+        Gets a WAP.QuotaRequest object - determining whether or not a deployment will succeed. 
+
+    .PARAMETER VMRoleSizeProfile
+        VM Size object to use as the basis.. Acquired via Get-WAPVMRoleVMSize.
+
+    .PARAMETER Size
+        VM Size name. Extra Small, Large, A7 etc.
+
+    .PARAMETER Subscription
+        WAP.SUbscription object to check. If left blank, returns the quota request for all of them. 
+
+    .EXAMPLE
+        PS C:\>$URL = 'https://publictenantapi.mydomain.com'
+        PS C:\>$creds = Get-Credential
+        PS C:\>Get-WAPToken -Credential $creds -URL 'https://sts.adfs.com' -ADFS
+        PS C:\>Connect-WAPAPI -URL $URL
+        PS C:\>$Sub = Get-WAPSubscription -Name 'MySubscription'
+        PS C:\>Get-WAPVMRoleQuotaRequest -Subscription $Sub -Size Large
+        OR
+        PS C:\>Get-WAPVMRoleQuotaRequest -Subscription $Sub -VMRoleSizeProfile (Get-WAPVMRoleVMSize -Name A7)
+
+    #>
+Param(
```

Comment:
cmdletbinding missing and same lower case and indent comments as the other function

What is the default parametersetname?